ci/check-rust-version.sh

```diff
@@ -1,49 +1,22 @@
-#!/bin/bash
+#!/bin/sh
```

Comment:
I would be in favor of using ``#!/usr/bin/env bash`` because posix sh/dash is a pain to work with and has many stupid quirks.

Reply:
`bash` is not available everywhere

Reply:
``apk add bash`` is a very simple fix for that and avoids all the quirks the many different implementations of /bin/sh have.

Reply:
`apk add bash` is not solution to every system without `bash`, and installing `bash` is also not a proper solution

Reply:
I don't know any system where bash is not available but rust works and I don't think there exists one but I know that ``/bin/sh`` is a different shell with different implantations on almost every platform I used so far and is a pain to support even with shellcheck:
- Debian has dash
- alpine has either busybox sh which has some more features than posix sh or also ash
- Arch has bash
- Darwin has something different, too
- NixOS has bash in the build sandbox and dash in the real system

For example if I would change the script on Arch I wouldn't even notice that I am using bash features while I break the script on Alpine or Debian. It is just so much more easier to just use bash which has the same feature set on all platforms except on darwin where the default bash is just to old.

---

Also making scripts POSIX compatible from bash is usually not worth the time because POSIX is just missing so many convenience features and features in general that it is often more a pain because you can't use so most things. best example is new test ``[[`` which is so much more usable compared to ``[`` and eliminates many of the quoting pitfalls.

Reply:
> I know that `/bin/sh` is a different shell with different implantations on almost every platform

It only matters if you write non-POSIX code. `bash` differences are even more painful than `/bin/sh`, because of breaking changes or bugs, and there is much more wrong with `bash` than that.

> Also making scripts POSIX compatible from bash is usually not worth the time because POSIX is just missing so many convenience features and features in general that it is often more a pain because you can't use so most things.

`bashisms` are terrible (as well as other GNU shenanigans), I just made script POSIX compliant and it isn't that hard, it took me about an hour or two, none of `bash` features are needed or useful to install deps or check rust version. The script is readable and not hard to understand

> For example if I would change the script on Arch I wouldn't even notice that I am using bash features while I break the script on Alpine or Debian. 

If I change something while still being POSIX compliant, nothing breaks nowhere :>

In the end, this is not an elaborate script doing "shell magic", skipping on `bash` is perfectly fine (and not only here imo but this is not a place to discuss that).
POSIX `sh` will still be the lowest common denominator.

Reply:
> bash differences are even more painful than /bin/sh, because of breaking changes or bugs, and there is much more wrong with bash than that.

I am not aware of any breaking changes or bugs in bash in the last years. I think you are just generally against bash and GNU in general. A binary path which is different on many systems and has many different implementations with widely varying feature sets is supposed to be more stable than one shell?

> I just made script POSIX compliant and it isn't that hard, it took me about an hour or two

Yeah, it took you one or two hours with the alternative being that you install one of the most widely used unix programs. Seams to be a not very fair tradeoff.

> If I change something while still being POSIX compliant

True but more people than you change things and many of them do not even have a fully posix shell on their system to check if things still work.